src/community/ogcapi/ogcapi-features/src/main/java/org/geoserver/api/features/CollectionsDocument.java

```diff
@@ -49,52 +49,53 @@ public CollectionsDocument(GeoServer geoServer, List<String> crsList) {
 
     @JacksonXmlProperty(localName = "Collection")
     public Iterator<CollectionDocument> getCollections() {
-        CloseableIterator<FeatureTypeInfo> featureTypes =
-                geoServer.getCatalog().list(FeatureTypeInfo.class, Filter.INCLUDE);
-        return new Iterator<CollectionDocument>() {
+        try (CloseableIterator<FeatureTypeInfo> featureTypes =
```

Comment:
Another case of "wrap and return", the featureTypes iterator must not be closed in this method, but when the returned iterator does it.

Reply:
Removed from the PR.